fp.js

```diff
@@ -31,4 +31,4 @@ var people = [
   {name: "Jessica", age: 34},
   {name: "Jasmine", age: 39}
 ]
-var peopleWhoseNamesHaveTheLetterS = people.filter(function(person){return person.name.match("s")});
+var peopleWhoseNamesHaveTheLetterS = people.forEach(function(person){return person.name.match("s")});
```

Comment:
In this case you do need to use push, because `.forEach` doesn't return anything of value here -- it just iterates through each item in an array and does whatever you tell it to in its callback. So the solution looks something like this...

```js
var peopleWhoseNamesHaveTheLetterS = []
people.forEach(function(person){
  if(person.name.match("s")){
    peopleWhoseNamesHaveTheLetterS.push(person)
  }
})
```